warehouse/ingest-core/src/main/java/datawave/ingest/data/config/XMLFieldConfigHelper.java

```diff
@@ -0,0 +1,567 @@
+            
+            if (!defaultsComplete) {
+                throw new IllegalStateException("Can't define a field without defaults - expected default tag before field tag");
+            }
+            
+            final int sz = attributes.getLength();
+            
+            String name = null;
+            boolean stored = this.defaultStored;
+            boolean indexed = this.defaultIndexed;
+            boolean reverseIndexed = this.defaultReverseIndexed;
+            boolean tokenized = this.defaultTokenized;
+            boolean reverseTokenized = this.defaultReverseTokenized;
+            String fieldType = this.defaultFieldType;
+            
+            for (int i = 0; i < sz; i++) {
+                final String qn = attributes.getQName(i);
+                final String lv = attributes.getValue(i);
+                
+                if (STORED.equals(qn)) {
+                    stored = Boolean.parseBoolean(lv);
+                } else if (INDEXED.equals(qn)) {
+                    indexed = Boolean.parseBoolean(lv);
+                } else if (REVERSE_INDEXED.equals(qn)) {
+                    reverseIndexed = Boolean.parseBoolean(lv);
+                } else if (TOKENIZED.equals(qn)) {
+                    tokenized = Boolean.parseBoolean(lv);
+                } else if (REVERSE_TOKENIZED.equals(qn)) {
+                    reverseTokenized = Boolean.parseBoolean(lv);
+                } else if ("name".equals(qn)) {
+                    name = lv;
+                } else if (INDEX_TYPE.equals(qn)) {
+                    fieldType = lv;
+                } else {
+                    throw new IllegalArgumentException("Unexpected attribute encounteded in: " + uri + " in 'field' tag: '" + qn + "'");
+                }
+            }
+            
+            if (name == null) {
+                throw new IllegalArgumentException("No field called 'name' specified");
+            } else if (!this.fieldHelper.addKnownField(name)) {
+                throw new IllegalArgumentException("Field " + name
+                                + " was already seen, check configuration file for duplicate entries (among fieldPattern, field tags)");
+            }
+            
+            if (stored) {
+                this.fieldHelper.addStoredField(name);
+            }
+            
+            if (indexed) {
+                this.fieldHelper.addIndexedField(name);
+            }
+            
+            if (reverseIndexed) {
+                this.fieldHelper.addReverseIndexedField(name);
+            }
+            
+            if (tokenized) {
+                this.fieldHelper.addTokenizedField(name);
+            }
+            
+            if (reverseTokenized) {
+                this.fieldHelper.addReverseTokenizedField(name);
+            }
+            
+            if (fieldType != null) {
+                if (this.ingestHelper != null) {
+                    this.ingestHelper.updateDatawaveTypes(name, fieldType);
```

Comment:
This refactor looks good to me. WRT our recent conversation, I'm guessing that the given XML schema _could_ support multiple normalizers for a single field by just making the **indexType** attribute a comma-delimited list(?) The updateDatawaveTypes method will parse your fieldType var that way at least. I guess a new junit in XMLFieldConfigHelperTest could test that hypothesis, if multiple normalizers per-field is something we need to continue to support

Reply:
Agreed, I think I'll tackle this in a separate ticket.